### ISSUE TITLE
DRBD 8.3.11 DRBD8Status regex failure

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#241]
Originally reported of Google Code with ID 185.

```
xx1:~/tmp/ganeti# gnt-cluster --version
gnt-cluster (ganeti v2.4.2) 2.4.2

xx1:~/tmp/ganeti# gnt-cluster version  
Software version: 2.4.2
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0

xx1:~# uname -a
Linux xx1 3.0.1-fai-server #1 SMP Fri Aug 12 17:51:32 CEST 2011 x86_64 GNU/Linux

Linux 3.0.2 vanilla changed a bit the output of /proc/drbd:


 3: cs:SyncSource ro:Primary/Secondary ds:UpToDate/Inconsistent A r-----
    ns:373888 nr:0 dw:0 dr:374088 al:0 bm:22 lo:7 pe:27 ua:7 ap:0 ep:1 wo:f oos:15358208
	[>....................] sync'ed:  2.4% (14996/15360)Mfinish: 0:04:08 speed: 61,736 (61,736) K/sec


The "M" between the size and finish fools the REGEXP that extract estimated time for drbd disk creation.

Attached a dead simple fix against git HEAD.

Bug also present un 2.4.2.
```
Originally added on 2011-08-18 14:39:48 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
It seems you forgot to attach the patch, at least I can't see it anywhere. We want to fix this bug in the 2.4.4 release.
```
Originally added on 2011-08-19 11:16:57 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
Oops, sorry.

This time I attached a patch.


```
Originally added on 2011-08-19 22:41:32 +0000 UTC.

Attachments: 
[patch_ganeti_bdev.diff](https://storage.googleapis.com/google-code-attachments/ganeti/issue-185/comment-2/patch_ganeti_bdev.diff)

**Comment by Ganeti-Issues-Migrator** [comment]
```
So after spending some time with the code base and some track downs, I could hunt this bug down to the drbd module included in the kernel. The bug was introduced in commit 4b0715f096 of torvalds linux git tree[1]. It's still unfixed in HEAD.

The "upstream" module of 8.3.11 however is not affected by this bug.

Thanks for your patch, I'll slightly adapt it and get it into Ganeti 2.4.4.

[1] http://git.kernel.org/?p=linux/kernel/git/torvalds/linux.git;a=commit;h=4b0715f09655e76ca24c35a9e25e7c464c2f7346
```
Originally added on 2011-08-22 13:58:23 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This was fixed in Ganeti 2.4.4. A patch sent to upstream still has been applied, but Ganeti now works with the broken output.

commit 7a380ddfc43c7092e0debf85b68a975728247813
Author: René Nussbaumer <rn@google.com>
Date:   Tue Aug 23 09:23:39 2011 +0200

    Fix a parsing issue with DRBD 8.3.11 in the Linux Kernel
    
    In the Linux kernel commit 4b0715f096 introduced a display bug into
    /proc/drbd which broke our regex.
    
    The bug was first introduced into Linux 2.6.39-rc1. This bug is still
    unfixed as of today.
    
    This patch adapt the regular expression to workaround this bug for the
    time being.
    
    Signed-off-by: René Nussbaumer <rn@google.com>
    Reviewed-by: Michael Hanselmann <hansmi@google.com>
```
Originally added on 2011-12-08 09:24:33 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Migrate old issues from 'fixed' to 'released'.
```
Originally added on 2012-10-22 09:16:17 +0000 UTC.

Changed State: Released
